### ISSUE TITLE
Gradient issues with `pad_packed_sequence` and/or `pack_padded_sequence`

### ISSUE THREAD

**Issue by egrefen** [facebookresearch/higher#38]
As flagged by @berlino and @Nithin-Holla in #27, there is an issue with either/both of `pad_packed_sequence` and/or `pack_padded_sequence` in `higher`, related to something which changed in pytorch v1.4.0.

Here is a minimal repro example, adapted from @Nithin-Holla's code:
```python
import higher
import torch
from torch.nn.utils.rnn import pack_padded_sequence, pad_packed_sequence

class Base(torch.nn.Module):
    def __init__(self, hidden=10):
        super(Base, self).__init__()
        self.hidden = hidden
        self.gru = torch.nn.GRU(input_size=50,
                          		hidden_size=self.hidden,
                          		num_layers=1,
                          		batch_first=True,
                          		bidirectional=True)
        self.linear = torch.nn.Linear(2 * self.hidden, 10)
        self.tanh = torch.nn.Tanh()

    def forward(self, input, input_len):
        packed = pack_padded_sequence(input, input_len, batch_first=True, enforce_sorted=False)
        hidden, _ = self.gru(packed)
        hidden, _ = pad_packed_sequence(hidden, batch_first=True)
        hidden = self.tanh(hidden)
        d = self.tanh(self.linear(hidden))
        return d

learner = Base()
loss_fn = torch.nn.CrossEntropyLoss()
lr = 0.1

support_x = torch.randn(size=(5, 10, 50))
support_y = torch.randint(0, 10, size=(5, 10))
support_len = torch.ones(5) * 10
query_x = torch.randn(size=(5, 10, 50))
query_y = torch.randint(0, 10, size=(5, 10))
query_len = torch.ones(5) * 10

optimizer = torch.optim.SGD(learner.parameters(), lr=lr)
with higher.innerloop_ctx(learner, optimizer) as (fmodel, diffopt):
    output = fmodel(support_x, support_len)
    loss = loss_fn(output.view(-1, 10), support_y.view(-1))
    diffopt.step(loss)
    output = fmodel(query_x, query_len)
    loss = loss_fn(output.view(-1, 10), query_y.view(-1))
    grads = torch.autograd.grad(loss, fmodel.parameters(time=0), allow_unused=True)

grad_is_none = [g is None for g in grads]

if any(grad_is_none):
	for id, grad in enumerate(grads):
	    if grad is None:
	        print('Gradient of param {}: {}'.format(id, grad))
	    else:
	    	print('Gradient of param {}: {}'.format(id, grad.shape))
else:
    print("All good!")
```

Running this code with pytorch 1.3.x (e.g. by switching to it using `pip install 'torch==1.3.1' --force-reinstall`) yields no null grads ("All good!"), while running with pytorch 1.4.0 (`pip install -U torch`) yields zero grads on all the module parameters for the segment of the forward pass between the calls to `pad_packed_sequence` and `pack_padded_sequence` (i.e. everything except the linear layer).

This, combined with the fact that unit tests introduced in #d52adfae08f675f91ec7f3722c2297ce636c5abe for `torch.nn.{LSTM,GRU,RNN}` (and their related cells) show that higher order gradients are propagating, means the most likely issue is something to do with `pad_packed_sequence` and/or `pack_padded_sequence`.

**Comment by egrefen** [comment]
@Nithin-Holla thanks for flagging. Someone else had sent this my way. It might be connected.

**Comment by egrefen** [comment]
Well, looks like it's not due to packing. Re-opening #27 and closing this one.
